src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -848,7 +867,7 @@ private void logConfiguration()
                value = dsProps;
             }
             value = (prop.contains("password") ? "<masked>" : value);
-            LOGGER.debug((prop + "................................................").substring(0, 32) + (value != null ? value : ""));
+            LOGGER.debug((prop + "................................................").substring(0, 32) + value);
```

Comment:
I didn't notice that.  If that is the case, we would prevent passwords with leading or trailing spaces.  Those values should not be trimmed.